packages/next-swc/crates/next-core/src/next_edge/context.rs

```diff
@@ -98,6 +99,35 @@ pub async fn get_edge_resolve_options_context(
         get_next_edge_import_map(project_path, ty, next_config, execution_context);
 
     let ty = ty.into_value();
+    let invalid_client_only_resolve_plugin = get_invalid_client_only_resolve_plugin(project_path);
+    let invalid_styled_jsx_client_only_resolve_plugin =
+        get_invalid_styled_jsx_resolve_plugin(project_path);
```

Comment:
These two calls could be moved into the branch below to avoid them for the types that don't need them